elasticsearch_dsl/document.py

```diff
@@ -277,25 +280,50 @@ def to_dict(self, include_meta=False):
         meta['_source'] = d
         return meta
 
-    def update(self, using=None, index=None, **fields):
+    def update(self, using=None, index=None, detect_noop=True, doc_as_upsert=False, **fields):
+        """ Update this document in Elasticsearch.
+
+        :arg using:
+        :arg index:
+        :arg detect_noop: check if values have changed to avoid unnecessary write operation. Default: True.
+        :arg doc_as_upsert: index document if it doesn't exist. Default: False.
+        :arg fields: kwargs of additional fields
+        """
         if not fields:
             raise IllegalOperation('You cannot call update() without updating individual fields. '
                                    'If you wish to update the entire object use save().')
+
         es = self._get_connection(using)
 
-        # update the data locally
-        merge(self._d_, fields)
+        # update given fields locally
+        if len(fields):
```

Comment:
hm, don't we want to be able to assign values to the DocType instance and then simply call .update()  to issue a full update?